homeassistant/components/nest/config_flow.py

```diff
@@ -103,19 +103,47 @@ def extra_authorize_data(self) -> Dict[str, str]:
 
     async def async_oauth_create_entry(self, data: dict) -> dict:
         """Create an entry for the SDM flow."""
+        assert self.is_sdm_api(), "Step only supported for SDM API"
         data[DATA_SDM] = {}
+
+        # Update existing config entry when in the reauth flow
+        existing_entries = self.hass.config_entries.async_entries(DOMAIN)
+        assert len(existing_entries) <= 1, "Unexpected config entries must be deleted"
+        if existing_entries:
+            self.hass.config_entries.async_update_entry(existing_entries[0], data=data)
+            return self.async_abort(reason="reauth_successful")
+
         return await super().async_oauth_create_entry(data)
 
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth upon an API authentication error."""
+        assert self.is_sdm_api(), "Step only supported for SDM API"
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Confirm reauth dialog."""
+        assert self.is_sdm_api(), "Step only supported for SDM API"
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema({}),
+                errors={},
+            )
+        return await self.async_step_pick_implementation(
+            user_input={"implementation": DOMAIN}
+        )
```

Comment:
I don't think we know what implementation needs the reauth.
```suggestion
        return await self.async_step_user()
```

Reply:
There was a little bit more to unravel here.  I think what this was doing was effectively doing before was skipping the code in self.setup_user, and invoking super().setup_user.  In particular, the "single_instance_allowed" block.

One approach here could have been to call super().setup_user, but that felt too subtle.  I instead introduced a reauth member variable to track this behavior change a little more directly.

I also moved up the duplicate config entry assertion to earlier in the flow so that it would fail at the start of the config flow rather than at the end, which seemed more straight forward.